src/Common/Uuid/UuidRegistry.php

```diff
@@ -115,9 +121,31 @@ public function createUuid()
     public function createMissingUuids()
     {
         // Empty should be NULL, but to be safe also checking for empty and null bytes
-        $resultSet = sqlStatementNoLog("SELECT `id` FROM `" . $this->table_name . "` WHERE `uuid` IS NULL OR `uuid` = '' OR `uuid` = '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0'");
+        $resultSet = sqlStatementNoLog("SELECT * FROM `" . $this->table_name . "` WHERE `uuid` IS NULL OR `uuid` = '' OR `uuid` = '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0'");
         while ($row = sqlFetchArray($resultSet)) {
-            sqlQuery("UPDATE " . $this->table_name . " SET `uuid` = ? WHERE `id` = ?", [$this->createUuid(), $row['id']]);
+            if (!$this->table_vertical) {
+                // standard case, add missing uuid
+                sqlQuery("UPDATE `" . $this->table_name . "` SET `uuid` = ? WHERE `id` = ?", [$this->createUuid(), $row['id']]);
+            } else {
+                // more complicated case where setting uuid in a vertical table. In this case need a uuid for each combination of table columns stored in $this->table_vertical array
+                $stringQuery = "";
+                $arrayQuery = [];
+                foreach ($this->table_vertical as $column) {
+                    $stringQuery .= " AND `" . $column . "` = ? ";
+                    $arrayQuery[] = $row[$column];
+                }
+                // First, see if it has already been set
+                $setUuid = sqlQuery("SELECT `uuid` FROM `" . $this->table_name . "` WHERE `uuid` IS NOT NULL AND `uuid` != '' AND `uuid` != '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0' " . $stringQuery, $arrayQuery)['uuid'];
+                if (!empty($setUuid)) {
+                    // Already set
+                    array_unshift($arrayQuery, $setUuid);
+                } else {
+                    // Not already set, so create
+                    array_unshift($arrayQuery, $this->createUuid());
+                }
+                // Now populate
+                sqlQuery("UPDATE `" . $this->table_name . "` SET `uuid` = ? WHERE 1 = 1 " . $stringQuery, $arrayQuery);
```

Comment:
sqlQueryNoLog

Reply:
I couldn't get `WHERE 1 = 1` part? :thinking:

Reply:
@yashrajbothra , the 1=1 (note it is always true) was basically a crutch since $stringQuery always started with a `AND`. I am going to rid of this since not "good" development :)